crates/node/src/kitchensink.rs

```diff
@@ -0,0 +1,280 @@
+use std::{
+    io::BufRead,
+    path::PathBuf,
+    process::{Child, Command, Stdio},
+    sync::atomic::{AtomicU32, Ordering},
+    time::Duration,
+};
+
+use alloy::{
+    network::EthereumWallet,
+    providers::{Provider, ProviderBuilder, ext::DebugApi},
+    rpc::types::{
+        TransactionReceipt,
+        trace::geth::{DiffMode, GethDebugTracingOptions, PreStateConfig, PreStateFrame},
+    },
+};
+
+use crate::Node;
+use revive_dt_config::Arguments;
+use revive_dt_node_interaction::{
+    EthereumNode, trace::trace_transaction, transaction::execute_transaction,
+};
+
+static NODE_COUNT: AtomicU32 = AtomicU32::new(0);
+
+#[derive(Debug)]
+pub struct KitchensinkNode {
+    id: u32,
+    base_directory: PathBuf,
+    substrate_binary: PathBuf,
+    eth_proxy_binary: PathBuf,
+    rpc_url: String,
+    wallet: EthereumWallet,
+    process_substrate: Option<Child>,
+    process_proxy: Option<Child>,
+}
+
+impl KitchensinkNode {
+    const BASE_DIRECTORY: &str = "kitchensink";
+    const SUBSTRATE_READY_MARKER: &str = "Running JSON-RPC server";
+    const ETH_PROXY_READY_MARKER: &str = "Running JSON-RPC server";
+    const BASE_SUBSTRATE_RPC_PORT: u16 = 9944;
+    const BASE_PROXY_RPC_PORT: u16 = 8545;
+
+    fn spawn_process(&mut self, _genesis: String) -> anyhow::Result<()> {
+        let substrate_rpc_port = Self::BASE_SUBSTRATE_RPC_PORT + self.id as u16;
+        let substrate_ws_port = substrate_rpc_port + 1;
```

Comment:
Isn't this wrap and conflict when the id is contiguous and more than 1 node is spawned?